setting-up-ocarina.md

```diff
@@ -0,0 +1,111 @@
+date_modified:  2021-07-05 15:01:00 +0000
+author: BioWilko
+maintainer: BioWilko
+---
+
+### 0. Register for an account via Majora
+[Read the instructions for how to register](register)
+
+### 1. Acquire permission to access a data view
+A data view is basically a list of metadata fields. If you are granted access to a data view, you can see all of the fields covered by that data view.
+Every data view is associated with a short name, which we sometimes call a `code` or Majora data view code (`mdv` code).
+
+Guidance on how to obtain access to a view is in progress, check back after the consortium agreement has been distributed.
+This document has been written ahead of time and we are not currently granting permissions to any restricted views. Do not attempt to contact anyone to arrange access at this time.
+
+### 2. Install the Ocarina client
+
+[Ocarina](https://github.com/SamStudio8/ocarina/tree/master/ocarina) is a command line tool that is used to connect to Majora and perform actions with elevated privileges that are not possible on the website.
+It is not terribly difficult to use, but ideally you will have used a command line tool before.
+
+You can install the latest version with the Python package manager. You'll probably want to install it into a conda environment on the shared node.
+
+```
+conda create -n mdv-ocarina python=3.7
+conda activate mdv-ocarina
+pip install git+https://github.com/samstudio8/ocarina.git
+```
+
+### 3. Register your Ocarina instance as an OAuth application
+
+Instructions to do this are located [here](oauth-app). Only the first section is necessary to set up ocarina but must be done before moving to the next step.
+
+### 4. Set your Ocarina credentials
+
+Create a JSON file in your home directory named `.ocarina`. Note the starting dot. [Download the example here](.ocarina).
+
+```
+{"MAJORA_DOMAIN": "https://majora.covid19.climb.ac.uk/", "MAJORA_USER": "your-username", "MAJORA_TOKEN": "OAUTH", "CLIENT_ID": "your-client-id", "CLIENT_SECRET": "your-client-secret"}
+```
+
+Where:
+* `MAJORA_DOMAIN` points to either the domain of the real, or test Majora. Do not miss the `/` at the end of the URL.
+* `MAJORA_USER` is your username on the appropriate Majora instance
+* `MAJORA_TOKEN` is set to `OAUTH` (note that you will not be able to use the `v2+` API without a rotating token)
+* `CLIENT_ID` is the Client ID of your registered application
+* `CLIENT_SECRET` is the Client Secret ID of your registered application
+
+OAuth has been the standard way to authenticate since September 2020 but legacy API keys are still supported, [instructions here.](getting-api-keys)
+
+### 5. Submit an ocarina query and generate an OAuth token
+
+Activate your Ocarina conda environment
+
+```
+conda activate mdv-ocarina
+```
+
+Attempt to access a biosample with the following command
+
+```
+ocarina --oauth get biosample --central-sample-id *Insert COG ID here*
+```
+
+You should then see the following message in your terminal
```

Comment:
Remove this example and just keep it simple:
"You should see a message asking you to 'request a grant'. Open the link (or copy it into your a browser on your local machine) to see an approval screen like this:"